nexus/src/app/background/region_replacement_driver.rs

```diff
@@ -0,0 +1,677 @@
+                                params: sagas::region_replacement_drive::Params {
+                                    serialized_authn: authn::saga::Serialized::for_opctx(opctx),
+                                    request,
+                                },
+                            }).await;
+
+                            match result {
+                                Ok(()) => {
+                                    ok += 1;
+                                }
+
+                                Err(e) => {
+                                    error!(&log, "sending region replacement drive request failed: {e}");
+                                    err += 1;
+                                }
+                            };
+                        }
+                    }
+                }
+
+                Err(e) => {
+                    error!(&log, "query for running region replacement requests failed: {e}");
+                }
+            }
+
+            // Complete region replacements that are done
+            match self.datastore.get_done_region_replacements(opctx).await {
+                Ok(requests) => {
+                    for request in requests {
+                        let Some(old_region_volume_id) = request.old_region_volume_id else {
+                            // This state is illegal!
+                            error!(
+                                &log,
+                                "old region volume id is None!";
+                                "request" => ?request,
+                            );
+
+                            continue;
+                        };
+
+                        let result = self.saga_request.send(sagas::SagaRequest::RegionReplacementFinish {
+                            params: sagas::region_replacement_finish::Params {
+                                serialized_authn: authn::saga::Serialized::for_opctx(opctx),
+                                region_volume_id: old_region_volume_id,
+                                request,
+                            },
+                        }).await;
+
+                        match result {
+                            Ok(()) => {
+                                ok += 1;
+                            }
+
+                            Err(e) => {
+                                error!(&log, "sending region replacement finish request failed: {e}");
+                                err += 1;
+                            }
+                        };
+                    }
+                }
+
+                Err(e) => {
+                    error!(&log, "query for done region replacement requests failed: {e}");
+                }
+            }
+
+            warn!(&log, "region replacement driver task done");
+
+            json!({
+                "region_replacement_driven_ok": ok,
+                "region_replacement_driven_err": err,
```

Comment:
1c31d3ea3 changes the task to return a struct with strings instead of information-destroying integer counters :)

Reply:
Is it just the tests that are using these results?

My concern was really that the results were not deterministic as to what happend
during the call to `drive_running_replacements_forward()`.  If there were multiple
ways a counter could get incremented (or a string added to a Vec), would the test
really know that the action it expected was what bumped the counter?  Maybe it is
true now, but some future day when I come in here and mess something up, the test
might still pass but it passes because it took a different path than expected.
  
So, my ask here is really that, for example when a test checks this:
```
assert_eq!(result.finish_invoked_ok.len(), 1);
```
It can only mean that a specific path has been taken.
Maybe we want a unique counter for each path (or unique something, idk really what)?

Does that request make sense?

Reply:
ah, I think I get it now - 0f0f26a6a changes to match against the expected strings that will be pushed in there.